web/src/compose.js

```diff
@@ -804,14 +800,35 @@ export function reset_compose_scheduling_state(reset_edit_state = true) {
 }
 
 function schedule_message_to_custom_date() {
-    const request = create_message_object();
-    const selected_send_later_time = popover_menus.get_selected_send_later_time();
-    request.content = `/schedule ${selected_send_later_time}\n` + request.content;
+    const compose_message_object = create_message_object();
+
+    const deliver_at = popover_menus.get_formatted_selected_send_later_time();
+    const send_later_time = popover_menus.get_selected_send_later_time();
+    const scheduled_delivery_timestamp = Math.floor(Date.parse(send_later_time) / 1000);
```

Comment:
This could maybe be a little cleaner, but I was mostly focused on keeping the web app implementation the same and just sending the new `scheduled_delivery_timestamp` to the server in the correct format.